src/EnergyPlus/HeatRecovery.cc

```diff
@@ -2190,11 +2192,52 @@ namespace HeatRecovery {
             //     Keep effectiveness between 0 and 1.0 ??
             //     HXOpSensEffect = MAX(MIN(HXOpSensEffect,1.0),0.0)
             //     HXOpLatEffect =  MAX(MIN(HXOpLatEffect,1.0),0.0)
-
-            //   The model should at least guard against negative numbers
-            ExchCond(ExNum).SensEffectiveness = max(0.0, ExchCond(ExNum).SensEffectiveness);
-            ExchCond(ExNum).LatEffectiveness = max(0.0, ExchCond(ExNum).LatEffectiveness);
-
+            bool static onetimewarningsensibleEff = false;
+            if (ExchCond(ExNum).SensEffectiveness < 0.0) {
+                //   The model should at least guard against negative numbers
+                ExchCond(ExNum).SensEffectiveness = 0.0;
+                if (!onetimewarningsensibleEff) {
+                    ShowWarningError("HeatExchanger:AirToAir:SensibleAndLatent =\"" + ExchCond(ExNum).Name + "\"" +
+                                     " sensible effectiveness is less than zero. Check the following inputs.");
+                    if (ExchCond(ExNum).SupInTemp < ExchCond(ExNum).SecInTemp) {
+                        ShowContinueError("...Sensible Effectiveness at 100% Heating Air Flow = " +
+                                          RoundSigDigits(ExchCond(ExNum).HeatEffectSensible100, 2));
+                        ShowContinueError("...Sensible Effectiveness at 75% Heating Air Flow = " +
+                                          RoundSigDigits(ExchCond(ExNum).HeatEffectSensible75, 2));
+                        ShowContinueError("...Sensible effectiveness reset to zero and the simulation continues.");
+                    } else {
+                        ShowContinueError("...Sensible Effectiveness at 100% Cooling Air Flow = " +
+                                          RoundSigDigits(ExchCond(ExNum).CoolEffectSensible100, 2));
+                        ShowContinueError("...Sensible Effectiveness at 75% Cooling Air Flow = " +
+                                          RoundSigDigits(ExchCond(ExNum).CoolEffectSensible75, 2));
+                        ShowContinueError("...Sensible effectiveness reset to zero and the simulation continues.");
+                    }
+                    onetimewarningsensibleEff = true;
+                }
+            }
+            bool static onetimewarninglatentEff = false;
```

Comment:
Rather than a static bool here, I'd add a couple of bools to the `ExchCond` struct. That way if more than one HX exhibits the problem, they can all be reported.

Reply:
OK. I will address your change request. @mjwitte Thanks.

Reply:
Also adding the following HX Air Volume Flow ratio as part of the error message. The flow ratio used to calculate the effectiveness.

```cpp 
ShowContinueError("...Heat Exchanger Air Volume Flow Ratio = " + RoundSigDigits(HXAirVolFlowRatio, 2));```